dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -318,7 +318,7 @@ protected void addMetadata( EventQueryParams params, List<DimensionItemKeywords.
                     .getOptionSetObject()
                     .getOptions()
                     .stream()
-                    .filter( opt -> grid.getRows().stream().anyMatch( r -> {
+                    .filter( opt -> opt != null && grid.getRows().stream().anyMatch( r -> {
                         Object o = r.get( columnIndex );
                         if ( o instanceof String )
                         {
```

Comment:
In the expression below `((String) o).equalsIgnoreCase( opt.getCode() )`, should we also check is if `opt.getCode()` is not null? I believe that the `code` is not a mandatory attribute.

Reply:
cleared